pkg/ccl/streamingccl/streamingest/stream_replication_e2e_test.go

```diff
@@ -350,4 +314,104 @@ func TestTenantStreamingProducerJobTimedOut(t *testing.T) {
 	<-time.NewTimer(3 * time.Second).C
 	require.Equal(t, [][]string{{"0"}},
 		c.destTenantSQL.QueryStr(t, "SELECT count(*) FROM d.t2 WHERE i = 3"))
+
+	// After resumed, the ingestion job paused on failure again.
+	c.destSysSQL.Exec(t, fmt.Sprintf("RESUME JOB %d", ingestionJobID))
+	jobutils.WaitForJobToPause(c.t, c.destSysSQL, jobspb.JobID(ingestionJobID))
+}
+
+func TestTenantStreamingPauseResumeIngestion(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	args := defaultTenantStreamingClustersArgs
+	c, cleanup := createTenantStreamingClusters(ctx, t, args)
+	defer cleanup()
+
+	producerJobID, ingestionJobID := c.startStreamReplication()
+
+	jobutils.WaitForJobToRun(c.t, c.srcSysSQL, jobspb.JobID(producerJobID))
+	jobutils.WaitForJobToRun(c.t, c.destSysSQL, jobspb.JobID(ingestionJobID))
+
+	var srcTime time.Time
+	c.srcSysSQL.QueryRow(t, "SELECT clock_timestamp()").Scan(&srcTime)
+	c.waitUntilHighWatermark(srcTime, jobspb.JobID(ingestionJobID))
+
+	c.compareResult("SELECT * FROM d.t1")
+	c.compareResult("SELECT * FROM d.t2")
+
+	// Pause ingestion.
+	c.destSysSQL.Exec(t, fmt.Sprintf("PAUSE JOB %d", ingestionJobID))
+	jobutils.WaitForJobToPause(t, c.destSysSQL, jobspb.JobID(ingestionJobID))
+	pausedCheckpoint := streamIngestionStats(t, c.destSysSQL, ingestionJobID).
+		ReplicationLagInfo.MinIngestedTimestamp
+	// Check we paused at a timestamp greater than the previously reached high watermark
+	require.True(t, srcTime.Before(pausedCheckpoint.GoTime()))
```

Comment:
I wonder if this will be flaky.  Is there anything that guarantees that we actually advance the checkpoint?  Perhaps should should be a Before or Equal check.